scrowl-io/owl-ui
# Sc 893 feature tooltip

### Short summary

In bootstrap-react, Tooltip and Popover are two variants of the Overlay component, so I've built our Tooltip component to accept an additional "popover" prop (boolean, false by default) to decide between the two. By mixing and matching the props/controls, we can have tooltips that appear on hover or popovers that appear on click. See Default-index.stories.tsx for an example of how to use the tooltip in a paragraph: the targetElement function currently returns a span, but can be modified to attach the overlay to any kind of element e.g. button or icon for popovers.

Currently working through some CSS issues with the positioning: upon clicking or hovering on the element, the tooltip/popover does not target the correct element and is positioned incorrectly. Forcing a re-render by changing the props seems to reload the CSS in a way that positions the overlay properly. 

---

### Test steps

1. In the Storybook Controls, select 'click' for the trigger, 'bottom' for the placement, and 'false' for the popover
2. Click on the bolded span to activate the overlay.
3. The tooltip appears far to the left of the targeted element
4. Back in controls, switch the popover property to 'true'
5. The tooltip changes into a popover, and is repositioned to the correct place below the element.
6. Switch the popover property back to 'false', and the tooltip remains in the correct spot.
